event_listener/dfusion_db/event_pusher.py

```diff
@@ -69,11 +78,34 @@ def update_accounts(event: Dict[str, Union[int, str, str, int]]):
         db.accounts.insert_one(new_account_record)
 
     elif transition_type == 1:  # Withdraw
-        pass
+
+        requested_withdraws = db.withdraws.find({'slot': event['slot']})
+
+        for deposit in requested_withdraws:
+            a_id = deposit['accountId']
+            t_id = deposit['tokenId']
+            amount = deposit['amount']
+
+            index = num_tokens * (a_id - 1) + (t_id - 1)
+            # Balances are stored as [b(a1, t1), b(a1, t2), ... b(a1, T), b(a2, t1), ...]
+            if balances[index] - amount >= 0:
+                _log.info("Decreasing balance of account {} - token {} by {}".format(a_id, t_id, amount))
+                balances[index] -= amount
+            else:
+                _log.info("Insufficient balance: account {} - token {} for amount {}".format(a_id, t_id, amount))
```

Comment:
Should we really leave the balance untouched, rather than making it negative?

I wonder if this can be attacked now, by posting a state transition that withdraws an infinite amount from an account, but leaves the root hash the same (which needs to be valid, in case there were only invalid withdraws).

As far as a driver can tell the transition would be valid (since all balances in the db still hash to the same state root) and they would likely not challenge the transition. However, if the balance was negative, they would know to challenge.

Reply:
I thought the point was that insufficient balance withdraw requests are untouched (as in not included in the bitmap). If we make the balances negative, then people would never be able to withdraw their actual balances in the future.

Suppose someone has a balance of 5 and request a withdraw of 10. We subtract their balance to -5 and don't process their request. Now they can never access their 5 again.

Please tell me if I am missing something. Keeping in mind that untouched request are simply encoded in the bitmap published to the claimable state.

Reply:
Ahhh, so we are trying to consider a case where none of the withdraw requests were valid and the state transition has an identical hash as the previous. I don't really see how this is a problem or should be challenged. I also don't yet understand the purpose of the "infinite" withdraw request as part of this example. Maybe you could elaborate. 

As of now, I am still unconvinced that the current logic is inconsistent.

Reply:
I think this discussion is somewhat similar with the bitmap discussion above. After thinking about it more, I think you are right that this should probably just mirror the driver's logic.